pkg/controller/operators/olm/operator.go

```diff
@@ -293,20 +304,54 @@ func newOperatorWithConfig(ctx context.Context, config *operatorConfig) (*Operat
 			return nil, err
 		}
 
-		// A separate informer solely for CSV copies. Object metadata requests are used
+		// A separate informer solely for CSV copies. Fields
+		// are pruned from local copies of the objects managed
 		// by this informer in order to reduce cached size.
-		gvr := v1alpha1.SchemeGroupVersion.WithResource("clusterserviceversions")
-		copiedCSVInformer := metadatainformer.NewFilteredMetadataInformer(
-			config.metadataClient,
-			gvr,
-			namespace,
-			config.resyncPeriod(),
-			cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
-			func(options *metav1.ListOptions) {
-				options.LabelSelector = v1alpha1.CopiedLabelKey
+		copiedCSVInformer := cache.NewSharedIndexInformerWithOptions(
+			pruning.NewListerWatcher(
+				op.client,
+				namespace,
+				func(opts *metav1.ListOptions) {
+					opts.LabelSelector = v1alpha1.CopiedLabelKey
+				},
+			),
+			&v1alpha1.ClusterServiceVersion{},
+			cache.SharedIndexInformerOptions{
+				ResyncPeriod: config.resyncPeriod(),
```

Comment:
I'm a bit worried about this resync period, if it is 30s as I suspect. That means we'll be re-listing all copied CSVs from the apiserver every 30s, which seems like it would be _much_ worse that re-listing all PartialObjectMetadata for copied CSVs every 30s. Neither is great though.

Do we actually need to resync that often? My feeling is that the resync interval should be something like 10h, but that we should also have a syncer tied to the workqueue such that we react to changes immediately.